sway-ir/src/optimize/memcpyopt.rs

```diff
@@ -824,59 +826,79 @@ enum CandidateKind {
     NonClobbered(Candidate),
 }
 
-// Is (an alias of) src_ptr clobbered on any path from load_val to store_val?
+/// Starting backwards from end_inst, till we reach start_inst or the entry block,
+/// is scrutiny_ptr (or an alias of it) stored to (i.e., clobbered)?/
+/// Also checks that there is no overlap (common symbols) b/w no_overlap_ptr and scrutiny_ptr.
 fn is_clobbered(
     context: &Context,
-    Candidate {
-        load_val,
-        store_val,
-        dst_ptr,
-        src_ptr,
-    }: &Candidate,
+    start_inst: &Value,
+    end_inst: &Value,
+    no_overlap_ptr: &Value,
+    scrutiny_ptr: &Value,
 ) -> bool {
-    let store_block = store_val.get_instruction(context).unwrap().parent;
+    let end_block = end_inst.get_instruction(context).unwrap().parent;
+    let function = end_block.get_function(context);
```

Comment:
`function` is used only to `get_entry_block`, which is additionally called in a loop. We can instead of the `function` have `entry_block` defined here and use it in the loop.  
```suggestion
    let entry_block = end_block.get_function(context).get_entry_block(context);
```